tests/cocotb/test_stream_tcdm.py

```diff
@@ -252,16 +262,217 @@ async def stream_tcdm_dut(dut):
     await snax_util.clock_and_wait(dut)
 
     # Start streamer again
-    await snax_util.reg_write(dut, CSR_START_STREAMER, 0)
+    await snax_util.reg_write(dut, CSR_START_STREAMER, 1)
     await snax_util.reg_clr(dut)
 
     # Wait for the rising edge of the valid
     # From here we can continuously stream for ever clock cycle
-    await RisingEdge(dut.stream2acc_data_0_valid_o)
+    await with_timeout(RisingEdge(dut.stream2acc_data_0_valid_o), 100, "ns")
     # Necessary for cocotb evaluation step
     await Timer(Decimal(1), units="ps")
 
     for i in range(LOOP_COUNT_0):
+        # Check if signal_valid is high, wait if not
+        await Timer(Decimal(1), units="ps")
+        while dut.stream2acc_data_0_valid_o.value != 1:
+            await snax_util.clock_and_wait(dut)
+
+        # Extract the data
+        read_stream_0 = int(dut.stream2acc_data_0_bits_o.value)
+
+        # Streamed data should be consistent
+        snax_util.comp_and_assert(wide_writer_golden_list[i], read_stream_0)
+        await snax_util.clock_and_wait(dut)
+
+    # reset the streamer
+    for _ in range(10):
+        await snax_util.clock_and_wait(dut)
```

Comment:
Why does it take 10 cycles to reset the streamer?

Reply:
No, but then it is very clear in the vcd where the first test ends / second test starts 😄

Reply:
Aahhhhhhh LOLOLOL ok 😆